schema/schema.go

```diff
@@ -348,36 +348,37 @@ func (schema *Schema) StateVersioning() bool {
 	return stateful
 }
 
-func (schema *Schema) SyncKeyTemplate() string {
+
+func (schema *Schema) SyncKeyTemplate() (syncKeyTemplate string, ok bool) {
 	syncKeyTemplateRaw, ok := schema.Metadata["sync_key_template"]
 	if !ok {
-		return ""
-	}
-	syncKeyTemplate, ok := syncKeyTemplateRaw.(string)
-	if !ok {
-		return ""
+		return
 	}
-	return syncKeyTemplate
+	syncKeyTemplate, ok = syncKeyTemplateRaw.(string)
+	return
 }
 
-func (schema *Schema) GenerateCustomPath(data map[string]interface{}) (string, error) {
-	tpl, err := pongo2.FromString(schema.SyncKeyTemplate())
-	if err != nil {
-		return "", err
+func (schema *Schema) GenerateCustomPath(data map[string]interface{}) (path string, err error) {
+	syncKeyTemplate, ok := schema.SyncKeyTemplate()
+	if !ok {
+		err = fmt.Errorf("Failed to read sync_key_template from schema %v", schema.URL)
+		return
 	}
-	path, err := tpl.Execute(pongo2.Context{}.Update(data))
+	tpl, err := pongo2.FromString(syncKeyTemplate)
 	if err != nil {
-		return "", err
+		return
```

Comment:
The err variable will be overridden in this case. Sample code to check:
`package main
 
import "fmt"
 
func main() {
  a, b := checkShadowing()
  fmt.Printf("%s, %s\n", a, b)
}

func checkShadowing() (path string, err error) {
	a, ok := get1()
	if !ok {
		err = fmt.Errorf("error not ok")
		return
	}
	b, err := get2(a)
	if err != nil {
		return
	}
	path, err = get3(b)
	return
}

func get1() (a string, ok bool) {
    return "aaa", true
}

func get2(a string) (b string, err error) {
    return a + "bbb", fmt.Errorf("error in get2")
}

func get3(b string) (c string, err error) {
    return b + c, fmt.Errorf("error in get3")
}` 

It returns ", error in get2", so desired error. This means that err is overridden.